File: 2023/day01/part2.ts
import { readFileSync } from "node:fs";

const lines = readFileSync("./input.txt", "utf-8")
	.replaceAll("\r", "")
	.split("\n");

const words = ["one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];

const calibration_values = lines.map(line => {
	const digits: string[] = [];
	for (let i = 0; i < line.length; i++) {
		if (/\d/.test(line.charAt(i))) {
			digits.push(line.charAt(i));
		} else {
			const word = words.find(w => line.startsWith(w, i));
			if (!word) continue;
			digits.push((words.indexOf(word) + 1).toString());
		}
	}

	return digits.at(0)! + digits.at(-1)!;
}).map(Number);

console.log(calibration_values.reduce((a, b) => a + b));

File: 2025/day01/part2.py
with open("input.txt") as file:
	lines = [line.rstrip() for line in file]

count = 0
dial = 50
for instruction in lines:
	direction = instruction[0]
	number = int(instruction[1:])
	for i in range(number):
		if direction == "L":
			dial -= 1
		else:
			dial += 1
		if dial % 100 == 0:
			count += 1
print(count)